File: HomeWork6_OOP/Core/MVP/Presenter.java
package HomeWork6_OOP.Core.MVP;

import HomeWork6_OOP.Core.Models.Worker;

import java.io.FileNotFoundException;

public class Presenter {
    Model model;
    View view;
    /**
     * Класс Presenter связывает работу View и Model для отображения актуальной информации и обработки данных от пользователя
     */
    public Presenter(View view, String path) {
        this.view = view;
        model = new Model(path);
    }

    /**
     * Сборка Worker
     */
    public void addWorker() {
        Worker worker = new Worker.WorkerBuilder(view.inputFullName())
                .setDateOfBirth(view.inputDateOfBirth())
                .setDivision(view.inputDivision())
                .setPosition(view.inputPosition())
                .setSalary(view.inputSalary())
                .build();
        model.saveWorker(worker);
    }

    public void showWorkers() throws FileNotFoundException {
        view.print(model.loadWorkers());
    }

    public void showMenu() {
        view.showMenu();
    }

    public String inputMenu() {
        return view.inputMenu();
    }

    public void exit() {
        view.exitMessage();
    }

    public void error() {
        view.errorMessage();
    }
}

File: HomeWork7_OOP/mvp/Presenter.py
from HomeWork7_OOP.mvp import Data_File, View_Console
from HomeWork7_OOP.models import Worker_Builder

# Класс Presenter связывает работу View и Model для отображения актуальной информации и обработки данных от пользователя
class Presenter:
    __data: Data_File
    __view: View_Console

    def __init__(self, view: View_Console, data: Data_File):
        self.__data = data
        self.__view = view

    def add_worker(self):
        worker = Worker_Builder(self.__view.input_full_name()) \
            .set_date_of_birth(self.__view.input_date_of_birth()) \
            .set_division(self.__view.input_division()) \
            .set_position(self.__view.input_position()) \
            .set_salary(self.__view.input_salary()) \
            .build()
        self.__data.save_worker(worker_list=worker.get_worker_list())

    def show_workers(self):
        self.__view.show_workers(self.__data.load_workers())

    def show_menu(self):
        self.__view.show_start_menu()

    def show_error(self):
        self.__view.show_error_message()

    def show_added(self):
        self.__view.show_added_message()

    def show_exit(self):
        self.__view.exit_message()

    def input_menu(self):
        return self.__view.input_menu()